pinecone/provider/index_resource_test.go

```diff
@@ -170,34 +166,13 @@ func testAccCheckIndexExists(resourceName string, index *pinecone.Index) resourc
 	}
 }
 
-func testAccDeleteIndex(resourceName string) resource.TestCheckFunc {
+func testAccCheckIndexDestroy(resourceName string) resource.TestCheckFunc {
 	return func(state *terraform.State) error {
-		indexResource, found := state.RootModule().Resources[resourceName]
-		if !found {
-			return fmt.Errorf("Resource not found in state: %s", resourceName)
+		_, found := state.RootModule().Resources[resourceName]
```

Comment:
I think all we actually need is to verify it's no longer in state and has been destroyed.